### PR TITLE
Change help text to use source/target terminology in service-mirror and healthchecks

### PR DESCRIPTION
**Pull request by kleimkuhler**
Change terminology from local/remote to source/target in service-mirror and
healthchecks help text.

This does not change any variable, function, struct, or field names since
testing is still improving

Signed-off-by: Kevin Leimkuhler <kevin@kleimkuhler.com>

### CODE REVIEW

File: cli/cmd/multicluster.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func buildMulticlusterInstallValues(opts *multiclusterInstallOptions) (*multiclu
 	}
 
 	if opts.gatewayProbePort == defaults.GatewayLocalProbePort {
-		return nil, fmt.Errorf("The probe port needs to be different from %d which is the local probe port", opts.gatewayProbePort)
+		return nil, fmt.Errorf("The probe port needs to be different from %d which is the source probe port", opts.gatewayProbePort)
```
Comment (adleong) [inline cli/cmd/multicluster.go:154]: how about "the multicluster probe port"?

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 2]
```diff
@@ -175,7 +175,7 @@ func NewRemoteClusterServiceWatcher(
 ) (*RemoteClusterServiceWatcher, error) {
 	remoteAPI, err := k8s.InitializeAPIForConfig(cfg, false, k8s.Svc)
 	if err != nil {
-		return nil, fmt.Errorf("cannot initialize remote api for cluster %s: %s", clusterName, err)
+		return nil, fmt.Errorf("cannot initialize target api for cluster %s: %s", clusterName, err)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:178]: "cannot initialize api for target cluster" ?

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 3]
```diff
@@ -271,7 +271,7 @@ func (rcsw *RemoteClusterServiceWatcher) cleanupOrphanedServices() error {
 
 	servicesOnLocalCluster, err := rcsw.localAPIClient.Svc().Lister().List(labels.Set(matchLabels).AsSelector())
 	if err != nil {
-		innerErr := fmt.Errorf("failed obtaining local services while GC-ing: %s", err)
+		innerErr := fmt.Errorf("failed obtaining source services while GC-ing: %s", err)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:274]: this whole message could use some work, it's not clear what GC-ing is, exactly.

"failed to list services in source cluster while cleaning up mirror services:"

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 4]
```diff
@@ -390,7 +390,7 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceDeleted(ev *RemoteSe
 // new gateway being assigned or additional ports exposed. This method takes care of that.
 func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceUpdated(ev *RemoteServiceUpdated) error {
 	serviceInfo := fmt.Sprintf("%s/%s", ev.remoteUpdate.Namespace, ev.remoteUpdate.Name)
-	rcsw.log.Debugf("Updating remote mirrored service %s/%s", ev.localService.Namespace, ev.localService.Name)
+	rcsw.log.Debugf("Updating target mirrored service %s/%s", ev.localService.Namespace, ev.localService.Name)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:393]: "Updating mirror service"

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 5]
```diff
@@ -490,7 +490,7 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceCreated(ev *RemoteSe
 	gatewaySpec, err := rcsw.resolveGateway(&ev.gatewayData)
 	if err == nil {
 		// only if we resolve it, we are updating the endpoints addresses and ports
-		rcsw.log.Debugf("Resolved remote gateway [%v:%d] for %s", gatewaySpec.addresses, gatewaySpec.incomingPort, serviceInfo)
+		rcsw.log.Debugf("Resolved target gateway [%v:%d] for %s", gatewaySpec.addresses, gatewaySpec.incomingPort, serviceInfo)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:493]: I think "Resolved gateway" is fine here

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 6]
```diff
@@ -543,7 +543,7 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteGatewayDeleted(ev *RemoteGa
 
 	var errors []error
 	if len(affectedEndpoints) > 0 {
-		rcsw.log.Debugf("Nulling %d endpoints due to remote gateway [%s/%s] deletion", len(affectedEndpoints), ev.gatewayData.Namespace, ev.gatewayData.Name)
+		rcsw.log.Debugf("Nulling %d endpoints due to target gateway [%s/%s] deletion", len(affectedEndpoints), ev.gatewayData.Namespace, ev.gatewayData.Name)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:546]: Again, I think "target gateway" is redundant and just "gateway" should suffice.

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 7]
```diff
@@ -645,7 +645,7 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteGatewayCreated(event *Remot
 }
 
 func (rcsw *RemoteClusterServiceWatcher) updateAffectedServices(gatewaySpec GatewaySpec, affectedServices []*corev1.Service) error {
-	rcsw.log.Debugf("Updating %d services due to remote gateway [%s/%s] update", len(affectedServices), gatewaySpec.gatewayNamespace, gatewaySpec.gatewayName)
+	rcsw.log.Debugf("Updating %d services due to target gateway [%s/%s] update", len(affectedServices), gatewaySpec.gatewayNamespace, gatewaySpec.gatewayName)
```
Comment (adleong) [inline controller/cmd/service-mirror/cluster_watcher.go:648]: ditto

File: pkg/healthcheck/healthcheck_multicluster.go [hunk 8]
```diff
@@ -118,21 +118,21 @@ func (hc *HealthChecker) multiClusterCategory() category {
 			},
 			{
 				description: "service mirror controller has required permissions",
-				hintAnchor:  "l5d-multicluster-local-rbac-correct",
+				hintAnchor:  "l5d-multicluster-source-rbac-correct",
```
Comment (adleong) [inline pkg/healthcheck/healthcheck_multicluster.go:121]: Are changing these hint anchors going to require a website change as well?

File: pkg/healthcheck/healthcheck_multicluster.go [hunk 9]
```diff
@@ -425,7 +425,7 @@ func (hc *HealthChecker) checkRemoteClusterConnectivity() error {
 
 			remoteAPI, err := k8s.NewAPIForConfig(clientConfig, "", []string{}, requestTimeout)
 			if err != nil {
-				errors = append(errors, fmt.Sprintf("* secret: [%s/%s] cluster: [%s]: could not instantiate remote api: %s", secret.Namespace, secret.Name, config.ClusterName, err))
+				errors = append(errors, fmt.Sprintf("* secret: [%s/%s] cluster: [%s]: could not instantiate target api: %s", secret.Namespace, secret.Name, config.ClusterName, err))
```
Comment (adleong) [inline pkg/healthcheck/healthcheck_multicluster.go:428]: could not instantiate api for target cluster

File: pkg/servicemirror/util.go [hunk 10]
```diff
@@ -26,14 +26,14 @@ func ParseRemoteClusterSecret(secret *corev1.Secret) (*WatchedClusterConfig, err
...
+		return nil, fmt.Errorf("secret should contain target cluster name as annotation %s", consts.RemoteClusterNameLabel)
 	}
 	if !hasDomain {
-		return nil, fmt.Errorf("secret should contain remote cluster domain as annotation %s", consts.RemoteClusterDomainAnnotation)
+		return nil, fmt.Errorf("secret should contain target cluster domain as annotation %s", consts.RemoteClusterDomainAnnotation)
 	}
 
 	if !hasL5dNamespace {
-		return nil, fmt.Errorf("secret should contain remote linkerd installation namespace as annotation %s", consts.RemoteClusterLinkerdNamespaceAnnotation)
+		return nil, fmt.Errorf("secret should contain target linkerd installation namespace as annotation %s", consts.RemoteClusterLinkerdNamespaceAnnotation)
```
Comment (adleong) [inline pkg/servicemirror/util.go:36]: target cluster linkerd installation

File: pkg/healthcheck/healthcheck_multicluster.go (hunk 8 above)
Comment (kleimkuhler) [reply pkg/healthcheck/healthcheck_multicluster.go:121]: They probably will. I think a follow-up change in the website is okay for this though. I will do a scan through the tutorial when it is ready and make those changes after this is merged.

Comment (zaharidichev) [reply pkg/healthcheck/healthcheck_multicluster.go:121]: There is a pending PR with all the website anchor changes that I will update once all of this lands. Have to incorporate the change that shows you the names of all alive/dead gateways that Alex did as well. So leave that with me :)

**Pull request merged**